### PR TITLE
Removed Action Name and Action Key and Action Description and introduced segment user id

### PR DESCRIPTION
**Pull request by gautamLL**
_This PR removes action_key, action_name and action_description. Clients would not need to send action_key now which was mandatory, livelike will create the key automatically on their end so that clients don't have to make any changes to their existing integration._

## Testing
- [✓]Tested locally using Actions Tester.
- [✓]Removed actions tests for action_key 
- [✓]Added tests for  custom_id, user_id and livelike_profile_id
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
Thanks for raising this PR @gautamLL , I'll schedule it for review.

File: packages/destination-actions/src/destinations/livelike-cloud/trackEvent/index.ts [hunk 1]
```diff
@@ -24,55 +28,61 @@ const action: ActionDefinition<Settings, Payload> = {
...
   defaultSubscription: 'type = "track"',
   fields: {
-    action_key: {
-      label: 'Action Key',
-      description:
-        'The unique key of Action. LiveLike will uniquely identify any event by this key. For example, `user-registration` could be a key for the action `USER REGISTRATION`.',
+    event_name: {
+      label: 'Event Name',
       type: 'string',
       required: true,
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/livelike-cloud/trackEvent/index.ts:34]: Hi @gautamLL , 
Looks like you are adding a new required field to this Integration. 
That's something that will cause all pre-existing instances of the Integration to break. 
If you have any customers using the Integration, their Integrations will stop working. 
The customers would then have to create new instances of the Integration. 

If you don't have any customers using the Integrations already then we should be OK to let this change through. 
Please confirm. 
Kind regards,
Joe

File: packages/destination-actions/src/destinations/livelike-cloud/trackEvent/index.ts [hunk 2]
```diff
@@ -24,55 +28,61 @@ const action: ActionDefinition<Settings, Payload> = {
...
+        '@path': '$.event'
       }
     },
-    action_description: {
-      label: 'Action Description',
-      description: 'The description of the Action.',
+    event_type: {
+      label: 'Event Type',
       type: 'string',
+      required: true,
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/livelike-cloud/trackEvent/index.ts:44]: note: this is adding another required field. If no customers are currently using the Integration we should be OK. Please confirm

**Review by joe-ayoub-segment — CHANGES_REQUESTED** [review]
hi @gautamLL , 
The only potential issue I see is that you're introducing some new 'required' fields. 
This will break all existing instances of the Integration. 
However if you don't have any customers using the Integration we can let this through. 

Please clarify. 
Kind regads,
Joe

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Partner confirmed that no customers are using this Integration. So OK to add new required fields.

**Comment by joe-ayoub-segment** [comment]
Hi @gautamLL ,
There was an issue which caused the CI to break. It wasn't something your change did, but you'll need to pull in the latest files from segmentio/action-destinations main branch before your PR will pass CI. Can you do this please?
Cheers,
Joe

**Comment by gautamLL** [comment]
Hi Joe
I just pull the latest changes and push that commit
Please let me know if any other thing is needed.
Thanks
Gautam Aggrawal

**Comment by joe-ayoub-segment** [comment]
Thanks @gautamLL , we'll get this out with our next deploy.

**Comment by joe-ayoub-segment** [comment]
Hi @gautamLL this PR has been deployed. Please let me know if you are happy with the change.

**Pull request merged**